### ISSUE TITLE
FortiusAnt does not pair with Tacx Genius

### ISSUE THREAD

**Issue by Johanboernl** [WouterJD/FortiusANT#424]
Hi, my name is Johan. 
I'm having trouble getting my Tacx training app to detect speed, cadence, and power data from my FortiusAnt 
application 

Strangely, it does detect  heart rate icon when I use it with the Tacx Genius bike trainer.

To give you more information, 
I'm using a Raspberry Pi 3 and an Ant+ dongle in conjunction with my Tacx Genius bike trainer, which is the 2021 model. 

Any ideas on what could be causing the issue?

![image](https://user-images.githubusercontent.com/89359541/224358145-5cb33513-6c32-4fbf-ac6d-c2f030910ba0.png)

![image](https://user-images.githubusercontent.com/89359541/224358830-2dcb035f-4c9a-4c1a-ba8f-60321413a781.png)

**Comment by WouterJD** [comment]
That's exactly why FortiusAnt does exist, Tacx wants to be closed

**Comment by Johanboernl** [comment]
So if I understand you correct 
You can't connect tacx cycling true Bluetooth with FortiusAnt

My FortiusAnt is connected to my tacx genius with ant+

Because my tacx genius doesn't have bluetooth

**Comment by WouterJD** [comment]
To avoid wrong conclusions, first try one of the leading applications like Trainer Road, Zwift or Rouvy and check whether they connect.
If they do - setup OK.
If they fo not - let's investigate.

If they do and TTS does not... I'd suspect TTS to have undocumented non-standard requirements.

**Comment by WouterJD** [comment]
Note that the pairing with the Geniusisnot completed (see FortiusAnt image) this can take a minute.

I think that only when pairing is complete, FortiusAnt is detectable. But I'd have to check the code for that

**Comment by Johanboernl** [comment]
Hi wouter

I have tested the tacx genius but it's stuck on pairing

When I disable ble  bless and node

I can pair en calibrate 

When you to on bless same behavior stuck on pairing tacx genius

**Comment by WouterJD** [comment]
Your FortiusAnt communicates
- ANT with the Genius
- BLE with the CTP (Zwift...)

These two processes are rather independant, try to get some logic how the pairing with the Genius works; I have no such system and cannot execute tests myself.

**Comment by Johanboernl** [comment]
Thank you for sharing. 
I have reviewed the documentation for the Tacx Genius and it states that pairing will only work if you power on the Tacx Genius first and then power up the Raspberry Pi. 

Currently, my only issue is that Swift is losing connection with the Raspberry Pi. 
When I start the pairing process, the Raspberry Pi is still waiting for approval of the connection.

It seems that Swift is losing connection with the Raspberry Pi after around four minutes of use, despite Bluetooth still being active. However, the dashboard of FortiusAnt shows the correct input of the ANT signal

what is the best solution for the raspberry pi

-node or bless

**Comment by WouterJD** [comment]
> what is the best solution for the raspberry pi
> 
> -node or bless

Functionally, there is no difference between nodejs or bless; I prefer bless because of the simplier architecture but that's invisible for the user.

**Comment by WouterJD** [comment]
What I would do from here is use thee simulation mode:

`FortiusAnt -s`
This simulates a trainer and produces the ANT- and/or BLE- signal to interface with Zwift.
I needed this myself for development (I do not want to cycle to be able to test the interface) but comes in handy for you as well.

Zwift connects to FortiusAnt and while executing the training as if somebody is cycling, you can have a coffee and/or look at FortiusAnt, the raspberry and your computer inspecting what happens.

**Comment by WouterJD** [comment]
On what system is Zwift running?

**Comment by Johanboernl** [comment]
Here's a possible revision for your text:

I've tested various applications, but they all seem to have the same issue: after around 5 minutes, I lose information from my Tacx Bluetooth connection, even though it's still connected. However, the FortiusAnt dashboard continues to show correct information.

Specifically, I've tested Zwift on desktop version 1.1.5 for Windows 10 and on iOS version 16.11 with app version 1.33.4. I've also tested Rouvy on iOS version 16.11 with app version 2.23.
